### PR TITLE
Add ignored_static to config

### PR DESCRIPTION
**Pull request by Raymi306**
**IMPORTANT: Please do not create a Pull Request adding a new feature without discussing it first.**

The place to discuss new features is the forum: <https://zola.discourse.group/>
If you want to add a new feature, please open a thread there first in the feature requests section.

Sanity check:

* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/getzola/zola/pulls) for the same update/change?

## Code changes
(Delete or ignore this section for documentation changes)

* [x] Are you doing the PR on the `next` branch?

If the change is a new feature or adding to/changing an existing one:

* [x] Have you created/updated the relevant documentation page(s)?
---
Issue #1925 
I need to do some manual testing for this yet. EDIT: done

### CODE REVIEW

**Comment by Raymi306** [comment]
Finally got around to manual testing, and my code is 100% not ignoring static files :weary:
Troubleshooting now. Below is how I set up my manual testing.
```
ignored_static = ["*ignored*"]
```
```
static/
├── folder
│   ├── ignored_2
│   └── regular
├── ignored_1
├── ignored_folder
│   └── regular
├── scripts
│   └── hello.js
└── site.css
```

edit: Was thankfully a simple fix

**Comment by Raymi306** [comment]
Any feedback on this PR @Keats ?

File: components/site/src/lib.rs [hunk 1]
```diff
@@ -591,7 +591,14 @@ impl Site {
...
         if self.static_path.exists() {
-            copy_directory(&self.static_path, &self.output_path, self.config.hard_link_static)?;
+            if let Some(gs) = &self.config.ignored_static_globset {
+                copy_directory_with_ignore_globset(
+                    &self.static_path,
+                    &self.output_path,
+                    self.config.hard_link_static,
+                    gs,
+                )?;
+            }
```
Comment (Keats) [inline components/site/src/lib.rs:609]: We would never copy the static folder if the user hasn't set ignored_static from what I'm understanding?

File: src/cmd/serve.rs [hunk 2]
```diff
@@ -475,6 +475,13 @@ pub fn serve(
     };
 
     let copy_static = |site: &Site, path: &Path, partial_path: &Path| {
+        // Do nothing if the file/dir is on the ignore list
+        println!("{:?}, partial: {:?}", path, partial_path);
```
Comment (Keats) [inline src/cmd/serve.rs]: That should be removed

**Review by Keats — CHANGES_REQUESTED** [review]

File: src/cmd/serve.rs (hunk 2 above)
Comment (Raymi306) [reply src/cmd/serve.rs]: whoops! will fix this immediately

File: components/site/src/lib.rs (hunk 1 above)
Comment (Raymi306) [reply components/site/src/lib.rs:609]: :facepalm: fixing now

File: components/utils/src/fs.rs [hunk 3]
```diff
@@ -140,6 +141,39 @@ pub fn copy_directory(src: &Path, dest: &Path, hard_link: bool) -> Result<()> {
     Ok(())
 }
 
+pub fn copy_directory_with_ignore_globset(
+    src: &Path,
+    dest: &Path,
+    hard_link: bool,
+    gs: &GlobSet,
```
Comment (Keats) [inline components/utils/src/fs.rs]: just pass an Option<GlobSet> to `copy_directory` instead of copying the function

**Review by Keats — CHANGES_REQUESTED** [review]

File: components/utils/src/fs.rs (hunk 3 above)
Comment (Raymi306) [reply components/utils/src/fs.rs]: done!

**Comment by Raymi306** [comment]
`error: package `time-core v0.1.1` cannot be built because it requires rustc 1.65.0 or newer, while the currently active rustc version is 1.64.0` --from Azure

**Comment by Keats** [comment]
Don't worry about the pinned version, we can increase it. The errors on the other CI are legit though

**Comment by Raymi306** [comment]
> Don't worry about the pinned version, we can increase it. The errors on the other CI are legit though

:grimacing:  didn't see the errors were different, will be more thorough next time.  Can't believe I forgot to actually build it, I must have been rushing.

Anyway, it builds now, will keep an eye on CI

**Review by Keats — APPROVED** [review]

**Comment by Keats** [comment]
Thanks!

**Comment by Jieiku** [comment]
I opened an issue report for this feature here: https://github.com/getzola/zola/issues/2677

**Pull request merged**